statsmodels/regression/linear_model.py

```diff
@@ -2912,16 +2924,38 @@ def summary2(
                       xname=xname, yname=yname, title=title)
         smry.add_dict(diagnostic)
 
+        etext = []
+
+        if not self.k_constant:
+            etext.append(
+                "R² is computed without centering (uncentered) since the \
+                model does not contain a constant."
+            )
+        if hasattr(self, 'cov_type'):
+            etext.append(self.cov_kwds['description'])
+        if self.model.exog.shape[0] < self.model.exog.shape[1]:
+            wstr = "The input rank is higher than the number of observations."
+            etext.append(wstr)
+
         # Warnings
         if eigvals[-1] < 1e-10:
             warn = "The smallest eigenvalue is %6.3g. This might indicate that\
             there are strong multicollinearity problems or that the design\
             matrix is singular." % eigvals[-1]
-            smry.add_text(warn)
+            etext.append(warn)
         if condno > 1000:
```

Comment:
In `summary` I put this in and `elif`. If we already have a very small eigenvalue, then we don't need to warn additionally about the condition number.